packages/features/webhooks/lib/WebhookService.ts

```diff
@@ -0,0 +1,40 @@
+import logger from "@calcom/lib/logger";
+import { safeStringify } from "@calcom/lib/safeStringify";
+
+import getWebhooks from "./getWebhooks";
+import sendOrSchedulePayload from "./sendOrSchedulePayload";
+
+const log = logger.getSubLogger({ prefix: ["[WebhookService] "] });
+
+/** This is a WIP. With minimal methods until the API matures and stabilizes */
+export class WebhookService {
+  private options = {} as Parameters<typeof getWebhooks>[0];
+  private webhooks: Awaited<ReturnType<typeof getWebhooks>> = [];
+  constructor(options: Parameters<typeof getWebhooks>[0]) {
+    return (async (): Promise<WebhookService> => {
+      this.options = options;
+      this.webhooks = await getWebhooks(options);
+      return this;
+    })() as unknown as WebhookService;
+  }
```

Comment:
```suggestion

  private constructor(options: Parameters<typeof getWebhooks>[0], webhooks: Awaited<ReturnType<typeof getWebhooks>>) {
    this.options = options;
    this.webhooks = webhooks;
  }

  static async create(options: Parameters<typeof getWebhooks>[0]): Promise<WebhookService> {
    const webhooks = await getWebhooks(options);
    return new WebhookService(options, webhooks);
  }
```

NIT: Can we do something like this. when calling new WebhooksService you dont get any indication on a TS level that its async. 

This way we have to call WebhookService.create(...) which returns Promise<WebhookService> and is a bit nicer on a type level. LMK what you think

Reply:
Yeah I've explored that option as well. My mind just wanted a shorter version since the Webhook class is useless unless you fetch the data. I see two possibilities. Allow to pass the already fetched webhooks. Or fetch them on init.

I'm fine with it either way.

Reply:
I'm fine either way too - they both work. Just food for thought

Reply:
Either way. The abstraction will make it easier to refactor later on.